utils/directories.cc

```diff
@@ -104,7 +105,10 @@ void verification_error(fs::path path, const char* fstr, Args&&... args) {
 // Verify that all files and directories are owned by current uid
 // and that files can be read and directories can be read, written, and looked up (execute)
 // No other file types may exist.
-future<> directories::do_verify_owner_and_mode(fs::path path, recursive recurse, int level) {
+// If a FilterFunc is provided, verifies only the paths that return true for the filter.
+template <typename FilterFunc>
+requires std::invocable<FilterFunc, fs::path&> && std::is_same_v<std::invoke_result_t<FilterFunc, fs::path&>, bool>
```

Comment:
Why a template?  Wouldn't a std::function work just as well? (and be simpler)

Reply:
Agreed. I have changed it to a std::function.